eZ/Publish/Core/Search/Legacy/Content/Common/Gateway/CriterionHandler/FullText.php

```diff
@@ -133,12 +133,7 @@ public function accept(Criterion $criterion)
      */
     protected function tokenizeString($string)
     {
-        return array_filter(
-            array_map(
-                'trim',
-                preg_split('(\\p{Z})u', strtr($string, '\'"', ''))
-            )
-        );
+        return preg_split('/[^\w|*]/u', $string, -1, PREG_SPLIT_NO_EMPTY);
```

Comment:
This fix proved to be more tricky than expected. The test case [`eZ\Publish\Core\Search\Legacy\Tests\Content\HandlerContentTest::testFullTextWildcardFilter`](https://github.com/ezsystems/ezpublish-kernel/blob/master/eZ/Publish/Core/Search/Legacy/Tests/Content/HandlerContentTest.php#L1052) showed that, when searching, we also need to support phrases with wildcards. 

Changing pattern from `/\W/u` to `/[^\w|*]/u` fixes it, but it also makes me wonder if there are more special cases I'm not aware of :) // cc @andrerom @pspanja 

Notes on the fix itself:
- behavior of `array_filter` is replaced by `PREG_SPLIT_NO_EMPTY` flag, 
- `trim` is not needed as it's covered by regex pattern,
- removal of double quotes is done earlier, by the `TransformationProcessor`.